### PR TITLE
Sscs 11740 update wants to attend

### PR DESCRIPTION
**Pull request by nickhill111**
[**Before creating a pull request make sure that:**](https://tools.hmcts.net/jira/browse/SSCS-11740)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java [hunk 1]
```diff
@@ -264,4 +265,34 @@ void givenAdjournmentNextHearingIsPaper_thenUpdateHearingChannel() {
         assertThat(callback.getCaseDetails().getCaseData().getLatestHearing().getValue().getHearingChannel()).isEqualTo(HearingChannel.PAPER);
     }
 
+    @DisplayName("When adjournment is enabled and case hearing type is face_to_face and Adjournment next hearing type is Paper "
+        + ", then case hearing type should updated the wants to attend to no.")
+    @Test
+    void givenAdjournmentNextHearingIsPaper_thenUpdateWantsToAttend() {
+        HearingDetails hearingDetails = new HearingDetails();
+        hearingDetails.setHearingChannel(HearingChannel.FACE_TO_FACE);
+        callback.getCaseDetails().getCaseData().setHearings(Arrays.asList(new Hearing(hearingDetails)));
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java:274]: callback.getCaseDetails().getCaseData() can be a local var

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java [hunk 2]
```diff
@@ -112,32 +116,49 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
...
+                }
+
+                log.info("Updating hearing type to {} and wants to attend to {}", hearingType, wantsToAttend);
+                appeal.getHearingOptions().setWantsToAttend(wantsToAttend);
+                appeal.setHearingType(hearingType);
+                sscsCaseData.getSchedulingAndListingFields().getOverrideFields().setAppellantHearingChannel(hearingChannel);
+            }
+
+            Hearing latestHearing = sscsCaseData.getLatestHearing();
+            if (nonNull(latestHearing) && nonNull(latestHearing.getValue())) {
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java:146]: I guess you need one for null latest hearing (and one for nonNull latestHearing + null getValue though I reckon this is a bit OTT)

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -264,4 +265,34 @@ void givenAdjournmentNextHearingIsPaper_thenUpdateHearingChannel() {
         assertThat(callback.getCaseDetails().getCaseData().getLatestHearing().getValue().getHearingChannel()).isEqualTo(HearingChannel.PAPER);
     }
 
+    @DisplayName("When adjournment is enabled and case hearing type is face_to_face and Adjournment next hearing type is Paper "
+        + ", then case hearing type should updated the wants to attend to no.")
+    @Test
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java:270]: Please change this to a ParameterizedTest with EnumSource HearingChannel

**Review by amoslv1 — CHANGES_REQUESTED** [review]
Couple of small changes

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java (hunk 1 above)
Comment (nickhill111) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java:274]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickhill111) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandler.java:146]: done

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java (hunk 3 above)
Comment (nickhill111) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java:270]: done

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java [hunk 4]
```diff
@@ -264,4 +265,34 @@ void givenAdjournmentNextHearingIsPaper_thenUpdateHearingChannel() {
...
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        assertThat(callback.getCaseDetails().getCaseData().getAppeal().getHearingOptions().getWantsToAttend()).isEqualTo(NO.getValue());
+        assertThat(callback.getCaseDetails().getCaseData().getAppeal().getHearingType()).isEqualTo(HearingType.PAPER.getValue());
+        assertThat(callback.getCaseDetails().getCaseData().getSchedulingAndListingFields().getOverrideFields().getAppellantHearingChannel()).isEqualTo(HearingChannel.PAPER);
+    }
+
+    @DisplayName("When adjournment is enabled and case hearing type is paper and Adjournment next hearing type is Face to Face "
+        + ", then case hearing type should updated the wants to attend to yes.")
+    @Test
```
[the remark this answers is not in the document]
Comment (nickhill111) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java:305]: removed as previous test now covers this

**Review by amoslv1 — APPROVED** [review]

**Pull request closed without merging**